invenio_rdm_records/resources/serializers/dcat/__init__.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Datacite to DCAT serializer."""
+
+from datacite import schema43
+from lxml import etree as ET
+from pkg_resources import resource_stream
+from werkzeug.utils import cached_property
+
+from invenio_rdm_records.resources.serializers import DataCite43XMLSerializer
+
+
+class DCATSerializer(DataCite43XMLSerializer):
```

Comment:
major: IMO inheritance is being broken here. The parent class is defined with `SimpleSerializer` and also the `BaseListSerializer`. The later one is allowing the proper serialization at the level of the first. This leads me to two questions:
- Why is `serialize_object_list` needed here if the implementation is identical to the formatter class (`SimpleSerializer`), see [here](https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/serializers/simple.py#L24)
- Could it be refactored so we also do not need `serialize_object`? If we move the call to `self.transform_with_xslt` inside `_etree_tostring` we could simply pass this function as encoder. This would require overwritting the constructor passing `encoder=self._etree_tostring`.

pseudocode:

```python
def __init__(self, **options):
        """Constructor."""
        super().__init__(encoder=self. _etree_tostring)

def _etree_tostring(self, record, **kwargs):
         root = self.transform_with_xslt(record, **kwargs)
         return ET.tostring(
             root,
             pretty_print=True,
             xml_declaration=True,
             encoding="utf-8",
         ).decode("utf-8")
```

Note that this will require a change in [`DataCite43XMLSerializer`](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/datacite/__init__.py#L29) to use options:

```python

def __init__(self, **options):
        """Constructor."""
        options = options or {"encoder": schema43.tostring}
        super().__init__(
            **options,
             ...
        )
```

It is a bit tedious (I'm sorry) but it will make use of inheritance and leave the code much cleaner/clear.